subsys/event_manager/event_manager.c

```diff
@@ -116,10 +117,9 @@ void * __weak event_manager_alloc(size_t size)
 	void *event = k_malloc(size);
 
 	if (unlikely(!event)) {
-		printk("Event Manager OOM error\n");
-		LOG_PANIC();
+		LOG_ERR("Event Manager OOM error\n");
 		__ASSERT_NO_MSG(false);
-		sys_reboot(SYS_REBOOT_WARM);
+		k_panic();
```

Comment:
I think we should leave reboot for release configurations to instantly reboot instead of being stuck waiting for watchdog here (or being stuck forever if there is no watchdog).